portal/components/customTokenDrawer/index.tsx

```diff
@@ -46,8 +46,20 @@ const canSubmit = ({
   !!l2customToken.l1Token &&
   isAddressEqual(l2customToken.l1Token, l1CustomToken.address)
 
-const getL1AddressValidity = (l1CustomToken: Token | undefined) =>
-  l1CustomToken ? 'this-address-is-valid' : 'this-address-is-not-valid'
+const getL1AddressValidity = function (
+  l1CustomToken: Token | undefined,
+  l2customToken: L2Token | undefined,
+  isL2PlaceholderData: boolean,
+) {
+  if (l2customToken && l2customToken.address && !l2customToken.l1Token) {
+    if (isL2PlaceholderData) {
+      return undefined
+    }
+    return 'l1-address-not-configured'
+  }
+
+  return l1CustomToken ? 'this-address-is-valid' : 'this-address-is-not-valid'
```

Comment:
The mix if nested `if`'s and conditionals make this difficult to read. What about something like this?

```suggestion
  if (l2customToken && l2customToken.address && !l2customToken.l1Token && isL2PlaceholderData) {
    return undefined
  } else if (l2customToken && l2customToken.address && !l2customToken.l1Token && !isL2PlaceholderData) {
    return 'l1-address-not-configured'
  } else if (l1CustomToken) {
    return 'this-address-is-valid'
  }
  return 'this-address-is-not-valid'
```

Reply:
As said, I think it makes the code much more clear. But I'm ok either way, as long as the logic is correct.